ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -407,32 +364,20 @@ void McapHandler::stop(
             stop_event_thread_nts_(event_lock);
         }
 
-        try
+        if (!configuration_.only_with_schema)
         {
-            if (!configuration_.only_with_schema)
-            {
-                // Adds to buffer samples whose schema was not received while running
-                add_pending_samples_nts_();
-            }
-            else
-            {
-                // Free memory resources
-                pending_samples_.clear();
-            }
-            dump_data_nts_();  // if prev_state == RUNNING -> writes buffer + added pending samples (if !only_with_schema)
-                               // if prev_state == PAUSED  -> writes added pending samples (if !only_with_schema)
+            // Adds to buffer samples whose schema was not received while running
+            add_pending_samples_nts_();
         }
-        catch (const std::overflow_error&)
+        else
         {
-            on_disk_full_(); // TODO: check if this is the right approach (could be here from a
-                             // previous callback execution, or normally on command reception)
+            // Free memory resources
+            pending_samples_.clear();
         }
+        dump_data_nts_();  // if prev_state == RUNNING -> writes buffer + added pending samples (if !only_with_schema)
+                            // if prev_state == PAUSED  -> writes added pending samples (if !only_with_schema)
 
-        // Close and rename MCAP file
-        close_file_nts_();
-
-        // Reset channels
-        channels_.clear();
```

Comment:
Double check if this is correct.

Reply:
What is the conclusion for this? It was not needed, or should not be there?

Reply:
I believe it was redundant. As it stands, there are two scenarios that call `stop`, an application level `stop` and an application level `suspend`. They both destroy the McapHandler, so the channels  are cleared automatically.

Reply:
But suspend does not destroy it right? That's why I was thinking it just shouldn't be there anymore. What I believe is that in principle it does not really matter since channels are created on the fly if there is none associated to a topic, but I wonder what are the implications for the size tracker.

Reply:
After thinking about it, I believe you are right. I'll clear the `channels` after stopping to avoid rewriting them on every new file after a suspend.